app/src/main/java/com/example/kotlinpps/activity/MainActivity.kt

```diff
@@ -1,20 +1,44 @@
 package com.example.kotlinpps.activity
 
+import android.content.Context
+import android.content.Intent
 import android.os.Bundle
 import androidx.appcompat.app.AppCompatActivity
-import com.example.kotlinpps.R
+import com.example.kotlinpps.GOOGLE_SIGN_IN
+import com.example.kotlinpps.SHARED_PREFERENCE
+import com.example.kotlinpps.database.UserRoomDatabase
+import com.example.kotlinpps.mvp.contract.MainActivityContract
 import com.example.kotlinpps.mvp.model.MainActivityModel
 import com.example.kotlinpps.mvp.presenter.MainActivityPresenter
 import com.example.kotlinpps.mvp.view.MainActivityView
+import com.facebook.stetho.Stetho
 import kotlinx.android.synthetic.main.activity_main.*
 
+
 class MainActivity : AppCompatActivity() {
-    private var presenter: MainActivityPresenter? = null
+    private var presenter: MainActivityContract.Presenter? = null
```

Comment:
Usign lateinit is better than initializing with a null value

Reply:
Done